fix(http): Add support for requesting body as URLSearchParams type

sera-wi opened the pull request:
There is no support for URLSearchParams type request in the current version. This commit adds serialize body function for URLSearchParams type.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: 36317

There is no support for URLSearchParams type request in the current version. When a user passes the body parameter with this type, Angular considers it as a random object. It ends up being omitted. 

## What is the new behavior?
This commit adds serialization for URLSearchParams type. It will handle URLSearchParams type request, it will not be omitted.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

sera-wi committed:
fix(http): Add support for requesting body as URLSearchParams type

There is no support for URLSearchParams type request in the current version. This commit adds serialize body function for URLSearchParams type.

alfaproject commented on packages/common/http/src/request.ts:
```diff
@@ -65,6 +65,15 @@ function isFormData(value: any): value is FormData {
+ *
+ * In some execution environments URLSearchParams is not defined.
+ */
+function isURLSearchParams(value: any): value is URLSearchParams {
```
```suggestion
function isURLSearchParams(value: unknown): value is URLSearchParams {
```

sera-wi committed:
fix(http): Update unit test for IE11

IE doesn't support URLSearchParams. There was no condition for an unsupported browser. It will fix unit test failure.

sera-wi committed:
fix(http): Change formatting

The If clause was not properly formatted. Bazel format command solved this issue to correct the file's format.

sera-wi committed:
fix(http): Update isURLSearchParams type to unknown

As per alfaproject's review, I changed isURLSearchParams input parameter value's type from any to unknown.

petebacondarwin reviewed (changes requested):
Hi @sera0731 - thanks for sending in this PR.

I note that @ajitsinghkaler also created a PR at #37852. But since yours came first I would like to help you to get this merged if you are still interested.

I think that serializing the value to a string is not the correct approach. Instead I think we should do similar to the other PR where we treat it like a special type that just gets passed through - like Blob and FormData.  The gives the most flexibility to the caller to then use the `URLSearchParams` API to extract what they want.

Can you update your code to do this and change the test accordingly?

alxhub commented:
👍 to @petebacondarwin's requests.

Additionally, this PR is currently 4 separate commits - it should probably be squashed into a single commit with the full implementation, and a commit message describing the changes being made and why in more detail.

petebacondarwin commented:
Reverting to #37852 due to a lack of response.
